controllers/complianceeventsapi/server.go

```diff
@@ -471,6 +505,95 @@ func parseQueryArgs(queryArgs url.Values, isCSV bool) (*queryOptions, error) {
 		}
 	}
 
+	parsed, err := setAuthorizedClusters(ctx, db, parsed, userConfig)
+	if err != nil {
+		// ErrNoAccess needs queryOptions
+		return parsed, err
+	}
+
+	return parsed, nil
+}
+
+// setAuthorizedClusters verifies that if a cluster filter is provided,
+// the user has access to this filter. If no cluster filter is provided,
+// it sets the cluster filter to all managed clusters the user has access to.
+// If the user has no access, then ErrNoAccess is returned.
+func setAuthorizedClusters(ctx context.Context, db *sql.DB, parsed *queryOptions,
+	userConfig *rest.Config,
+) (*queryOptions, error) {
+	unAuthorizedClusters := []string{}
+
+	// Get all managedCluster rules
+	allRules, err := getManagedClusterRules(userConfig, nil)
+	if err != nil {
+		return parsed, err
+	}
+
+	if slices.Contains(allRules["*"], "get") || slices.Contains(allRules["*"], "*") {
+		return parsed, nil
+	}
+
+	tempClusterIDs := parsed.Filters["clusters.cluster_id"]
+	clusterIDs := make([]string, len(tempClusterIDs))
+
+	// Deep copy parsed.Filters["clusters.cluster_id"]
+	copy(clusterIDs, tempClusterIDs)
+
+	// Reset clusters.cluster_id
+	parsed.Filters["clusters.cluster_id"] = []string{}
+	// Convert id to name
+	for _, id := range clusterIDs {
+		clusterName, err := getClusterNameFromID(ctx, db, id)
+		if err != nil {
+			if errors.Is(err, sql.ErrNoRows) {
+				// Filter out invalid cluster IDs from the query
+				continue
```

Comment:
After thinking about this, I think this allows a user to bypass all authorization checks and have everything returned if they provide an invalid cluster ID.

I'll fix this in an upcoming PR.